hashicorp/consul-k8s
# Endpoints controller with tests

The endpoints controller and test code is ready for review. 

The only things expected to change at this point are code and tests in handler (to remove always setting the service-name annotation) and inject-connect (to get the command unit tests to pass since the manager takes a kubeconfig that's hard to fake out).

Changes in this PR:
* supports creating, updating and deleting endpoints
  * with different consul service name (using the service-name pod annotation) 
* updating endpoints supports adding and removing addresses from the endpoints object.

Todos:
* testing with new connect-init command
* testing end to end

Future PR:
* delete default annotation for service name (to not cause rebase issues with webhook refactor PR lined up after this one)
* inject-connect tests-- the webhook PR has a fix for one of these tests, and the other test is signal handling that would likely be refactored out. There's a TODO for this in the code.
* add metrics config to proxy registration (this is going to a future PR so it won't conflict with the upcoming webhook refactor PR)
* deleting old service registration

Co-authored-by: Iryna Shustava <iryna@hashicorp.com>
Co-authored-by: Kyle Schochenmaier <kyle.schochenmaier@hashicorp.com>

How I've tested this PR:
* Unit tests for endpoints controller

How I expect reviewers to test this PR:


Checklist:
